### PR TITLE
S3: Adding in damage logic for straightforward sections

### PR DESCRIPTION
**Pull request by shark20061**
Damage logic for the less complicated rooms that need it

Boiler Access
Garbage Chute
Lava Maze
Sova Processing
Sova Suite

All of this logic adds only one or two damage run options, since there aren't major obstacles for these runs

### CODE REVIEW

File: randovania/games/fusion/logic_database/Sector 3 PYR.txt [hunk 1]
```diff
@@ -643,24 +650,32 @@ Extra - room_id: [19]
...
-          Gravity Suit and Screw Attack
-          # Heat Run Logic
-          Varia Suit
+          Screw Attack
+          Any of the following:
+              # Heat Run logic (need Gravity to do Screw Attack in the lava)
+              # Gravity - Lava and Heat Damage
+              Gravity Suit and Damage Runs (Intermediate) and Lava Damage ≥ 110
+              # Varia and Gravity - No Damage
+              Gravity Suit and Varia Suit
```
Comment (JeffGainsNGames) [inline randovania/games/fusion/logic_database/Sector 3 PYR.txt]: common requirement of gravity can be moved into `all of`
```
all of
 screw and grav
 any of
   damage runs
   varia
```

File: randovania/games/fusion/logic_database/Sector 3 PYR.txt [hunk 2]
```diff
@@ -674,18 +689,29 @@ Extra - room_id: [19]
...
-          Gravity Suit and Screw Attack
-          # Heat Run logic
-          Varia Suit
+          Screw Attack
+          Any of the following:
+              # Heat Run logic (need Gravity to do Screw Attack in the lava)
+              # Gravity - Lava and Heat Damage
+              Gravity Suit and Damage Runs (Intermediate) and Lava Damage ≥ 110
+              # Varia and Gravity - No Damage
+              Gravity Suit and Varia Suit
```
Comment (JeffGainsNGames) [inline randovania/games/fusion/logic_database/Sector 3 PYR.txt]: see above

File: randovania/games/fusion/logic_database/Sector 3 PYR.txt (hunk 1 above)
Comment (shark20061) [reply randovania/games/fusion/logic_database/Sector 3 PYR.txt]: Fixed

File: randovania/games/fusion/logic_database/Sector 3 PYR.txt (hunk 2 above)
Comment (shark20061) [reply randovania/games/fusion/logic_database/Sector 3 PYR.txt]: Fixed

**Pull request merged**